File: app/src/main/java/net/mbonnin/arcanetracker/GameAdapter.java
package net.mbonnin.arcanetracker;

import android.support.v7.widget.RecyclerView;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;

import net.mbonnin.arcanetracker.model.GameSummary;

import java.util.List;

public class GameAdapter extends RecyclerView.Adapter {

    private final List<GameSummary> mGameList;

    public interface Listener {
        void onClick(GameSummary summary);
    }
    public GameAdapter(List<GameSummary> gameList) {
        mGameList = gameList;
    }

    @Override
    public RecyclerView.ViewHolder onCreateViewHolder(ViewGroup parent, int viewType) {
        View view = LayoutInflater.from(parent.getContext()).inflate(R.layout.game_view, parent, false);
        return new GameViewHolder(view);
    }

    @Override
    public void onBindViewHolder(RecyclerView.ViewHolder holder, int position) {
        ((GameViewHolder)holder).bind(mGameList.get(position), position);
    }

    @Override
    public int getItemCount() {
        return mGameList.size();
    }

}

File: app/src/main/kotlin/net/mbonnin/arcanetracker/ui/my_games/GameAdapter.kt
package net.mbonnin.arcanetracker.ui.my_games

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.squareup.sqldelight.runtime.coroutines.asFlow
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.GlobalScope
import kotlinx.coroutines.flow.collect
import kotlinx.coroutines.launch
import net.mbonnin.arcanetracker.R
import net.mbonnin.arcanetracker.sqldelight.Game
import net.mbonnin.arcanetracker.sqldelight.mainDatabase

class GameAdapter() : RecyclerView.Adapter<RecyclerView.ViewHolder>() {
    var mGameList = emptyList<Game>()

    fun updateGames(gameList: List<Game>) {
        mGameList = gameList
        notifyDataSetChanged()
    }
    override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): RecyclerView.ViewHolder {
        val view = LayoutInflater.from(parent.context).inflate(R.layout.game_view, parent, false)
        return GameViewHolder(view)
    }

    override fun onBindViewHolder(holder: RecyclerView.ViewHolder, position: Int) {
        (holder as GameViewHolder).bind(mGameList[position], position)
    }

    override fun getItemCount(): Int {
        return mGameList.size
    }
}
